src/components/messages/conversation-preview.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC } from "react";
+import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import { Conversation, Profile } from "../../models";
+
+const styles = StyleSheet.create({
+  container: {
+    height: 65,
+    justifyContent: "center",
+    marginVertical: 2,
+  },
+  infoContainer: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+    paddingHorizontal: 5,
+  },
+  picture: {
+    height: 60,
+    width: 60,
+    borderRadius: 60 / 2,
+  },
+  titleAndMessageContainer: {
+    flexDirection: "column",
+    alignItems: "flex-start",
+    justifyContent: "center",
+    width: "100%",
+    paddingLeft: 10,
+  },
+  title: {
+    fontWeight: "bold",
+    fontSize: 17,
+    marginBottom: 4,
+  },
+  messageAndTime: {
+    flexDirection: "row",
+    justifyContent: "space-between",
+  },
+  message: {
+    fontSize: 14,
+    color: "#9E9E9E",
+  },
+  date: {
+    fontSize: 14,
+    color: "#9E9E9E",
+    paddingLeft: 10,
+  },
+});
+
+type Props = {
+  conversation: Conversation;
+  profile: Profile;
+};
+
+export const ConversationPreview: FC<Props> = (props: Props) => {
+  const getTitle: () => string = () => {
+    if (props.conversation.title) return props.conversation.title;
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    let title = "";
+    profiles.forEach((profile, index) => {
+      title += profile.firstName + " " + profile.lastName;
+      if (index !== profiles.length - 1) {
+        title += ", ";
+      }
+    });
+    return title;
+  };
+
+  const getPicture: () => string = () => {
+    const profiles = props.conversation.users.filter(
+      (user) => user.id !== props.profile.id
+    );
+    return profiles[0].profilePicture;
+  };
+
+  const getLastMessage: () => string = () => {
+    let lastMessage =
+      props.profile.id ===
+      props.conversation.messages[props.conversation.messages.length - 1].sender
+        .id
+        ? "Vous: "
+        : props.conversation.users.length > 2
+        ? props.conversation.messages[props.conversation.messages.length - 1]
+            .sender.firstName + ": "
+        : "";
+    lastMessage +=
+      props.conversation.messages[props.conversation.messages.length - 1]
+        .content;
+    lastMessage =
+      lastMessage.length >= 43
+        ? lastMessage.substring(0, 43 - 3) + "..."
+        : lastMessage;
```

Comment:
There's a cleaner way to do this in CSS, see https://developer.mozilla.org/fr/docs/Web/CSS/text-overflow

Reply:
Use this instead https://reactnative.dev/docs/text#numberoflines (because V8 isn't the render engine)

Reply:
We are removing characters and not lines

Reply:
I now use numberOfLines props but the date is stucked at the right of the screen (but it looks nice)

Reply:
I didn't know that if you specify 1 line it will make an ellipsis to the end of the line 😮